platform/MCU/nRF52840/src/flash_mal.c

```diff
@@ -416,8 +417,9 @@ int FLASH_CopyMemory(flash_device_t sourceDeviceID, uint32_t sourceAddress,
         if (!(flags & MODULE_DROP_MODULE_INFO) && (flags & MODULE_VERIFY_MASK)) {
             // We may only check the module header if we've been asked to verify it
             uint32_t infoOffset = 0;
-            const module_info_t* info = FLASH_ModuleInfo(sourceDeviceID, sourceAddress, &infoOffset);
-            if (info->flags & MODULE_INFO_FLAG_DROP_MODULE_INFO) {
+            module_info_t info = {};
+            FLASH_ModuleInfo(&info, sourceDeviceID, sourceAddress, &infoOffset);
```

Comment:
Check return value here and propagate `FLASH_ACCESS_RESULT_ERROR` in case of failure.